zamba/data/video.py

```diff
@@ -325,6 +325,41 @@ def validate_total_frames(cls, values):
         return values
 
 
+def get_cached_array_path(config, vid_path, cache_dir):
+    """Get the path to where the cached array would be, if it exists.
+
+    config: dictionary or VideoLoaderConfig
+    vid_path: string path to the video, or Path, or S3Path
+    cache_dir: string directory name, or Path, or S3Path
+
+    returns: Path object to the cached data
+    """
+    if isinstance(config, VideoLoaderConfig):
+        # NOTE: config.dict() and dict(config) are equal, but their string representations are not
+        # so we should use config.dict() everywhere
```

Comment:
are there other places in the code where you see this? seems either worth fixing or worth filing an issue for so we remember to do it

Reply:
The issue is that BaseModel.dict() is recursive, so sub-models are converted to dictionaries, too. 

https://pydantic-docs.helpmanual.io/usage/exporting_models/#modeldict

That is arguably the right behavior, so I don't think we want to change it.

There are probably not many places where this is an issue. It mostly came up in testing.

But it's an argument in favor of a stricter signature for `load_video_frames`, like requiring a VideoLoaderConfig and not accepting a dictionary or kwargs.

Reply:
I am working now on moving the cache to the right key, and I have been banging my head on this issue repeatedly. I think `get_cached_array_path` should take either a dictionary or a `VideoLoaderConfig`; supporting both is very error prone. I am leaning toward requiring `VideoLoaderConfig`. Thoughts?

Reply:
the original reason we set up `load_video_frames` to take in the config was so that we only had to document the options in one place (VideoLoaderConfig). if we support the kwargs, then we have to replicate all the documentation and make sure the options match between the two if one is added/removed.

but then we added support for a dictionary i think for ease-of-use (eg. saves a user from needing to import VideoLoaderConfig) and maybe to reduce dependency conflicts. i remember something vague about `load_video_frames` being helpful as a generic video loader, but there was some problem with because `VideoLoaderConfig` includes running Megadetector (if that option is chosen) and that requires having all the yolox dependencies. i think maybe there was something nice about being able to just specify the resizing options directly `load_video_frames(filepath, height=420, width=420)` and just use that pathway. anyway, that is all very hand-wavy because I don't remember the actual problem and so likely isn't something we should design around. 

looking through the code, i think in `zamba` we always pass in a config and don't make use of the dict (this was more when we were working in notebooks). so all that said, i'm inclined to agree with you allen

Reply:
Ok, I made that change and so far it is looking like a good move. Just pushed a commit with the changes.